packages/directory/src/directory-fs.ts

```diff
@@ -213,7 +216,15 @@ export function createDirectoryFs(fs: IFileSystem, directoryPath: string): IFile
       return fs.writeFile(path === '' ? '' : resolveFullPath(path), ...args);
     } as IBaseFileSystemCallbackActions['writeFile'],
     readlink(path, callback) {
-      return fs.readlink(resolveFullPath(path), callback);
+      return fs.readlink(resolveFullPath(path), function (e, path) {
+        callback(e, path ? resolveScopedPath(relative(directoryPath, path)) : path);
```

Comment:
@AviVahl I need to fix the path that returns from the readlink to be scoped to the directory

Reply:
your test found it :+1:

Reply:
wait, but the `path` is relative itself and needs to be `fs.resolve()`'ed

Reply:
the `path` returned is absolute. Then I need to make it relative to the `directoryPath` so it will be scoped. But relative returns me without a `/` in the beginning